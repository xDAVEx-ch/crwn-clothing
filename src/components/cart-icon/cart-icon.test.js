import React from 'react';
import { shallow } from 'enzyme';

import CartIcon from './cart-icon.component';

describe('Cart icon component', () => {

    let wrapper;
    let mockToggleCartDropdown;

    beforeEach(() => {
        mockToggleCartDropdown = jest.fn();
        let mockItemsCount = 7;

        let mockProps = {
            toggleCartHidden: mockToggleCartDropdown,
            itemCount: mockItemsCount
        }

        wrapper = shallow(<CartIcon.WrappedComponent {...mockProps} />);
    });

    test('should render icon component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should toggle the cart dropdown component when is clicked', () => {
        wrapper.find('CartIconContainer').simulate('click');
        expect(mockToggleCartDropdown).toHaveBeenCalled();
    });

    test('should display number of items', () => {
        const itemsNumber = parseInt(wrapper.find('ItemCountContainer').text());
        expect(itemsNumber).toBe(7);
    })
})