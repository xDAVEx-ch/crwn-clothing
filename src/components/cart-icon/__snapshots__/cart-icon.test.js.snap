// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart icon component should render icon component 1`] = `ShallowWrapper {}`;
